packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -717,8 +752,27 @@ export class GarbageCollector implements IGarbageCollector {
             };
         }
 
+        const newSerializedSummaryState = JSON.stringify(generateSortedGCState(gcState));
         const builder = new SummaryTreeBuilder();
-        builder.addBlob(`${gcBlobPrefix}_root`, JSON.stringify(gcState));
+        builder.addBlob(gcBlobRootKey, newSerializedSummaryState);
+
+        /**
+         * As an optimization if the GC tree hasn't changed and we're tracking the gc state, return a tree handle
+         * instead of returning the whole GC tree. If there are changes, then we want to return the whole tree.
+        */
+        if (this.trackGCState) {
+            this.pendingSerializedSummaryState = newSerializedSummaryState;
+            if (
+                this.latestSerializedSummaryState !== undefined &&
+                this.latestSerializedSummaryState === newSerializedSummaryState &&
+                !fullTree &&
+                trackState
+            ) {
+                const gcSummaryPath = `/${gcTreeKey}`;
```

Comment:
If this condition is true, all this returns is an `ISummarizeResult` with the `summary` as `ISummaryHandle`. Take a look at how summarizer node returns this in its `summarize` function.

Reply:
You don't need to create a builder in this case. Only create the builder if you want to write a tree.